src/NUnitConsole/nunit-console/Utilities/TeamCityEventHandler.cs

```diff
@@ -45,10 +46,20 @@ public TeamCityEventHandler(TextWriter outWriter)
             _outWriter = outWriter;
         }
 
-        public void TestStarted(XmlNode node)
+        public void SuiteStarted(XmlNode testEvent)
         {
-            string name = node.GetAttribute("fullname");
-            string testId = node.GetAttribute("id");
+            _asemblyResolver.RegisterSuite(testEvent.GetAttribute("id"), testEvent.GetAttribute("fullname"));
+        }
+
+        public void SuiteFinished(XmlNode testEvent)
+        {
+            _asemblyResolver.UnregisterSuite(testEvent.GetAttribute("id"), testEvent.GetAttribute("fullname"));
+        }
+
+        public void TestStarted(XmlNode testEvent)
+        {
+            string testId = testEvent.GetAttribute("id");
+            string name = this.EnrichTestName(testId, testEvent.GetAttribute("fullname"));
```

Comment:
Since users can arbitrarily change the name of a test, fullname attribute will not necessarily match the class and method. I think the class portion will usually be correct, but parsing it out is quite difficult if the name contains string arguments and those strings contain certain characters. It would be much simpler to use the classname attribute, which is provided for the purpose when looking up the assembly.